feg/gateway/sbi/sbi_log_middleware.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2021 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sbi
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/labstack/echo/v4"
+)
+
+type Logger interface {
+	// Skipper returns true to skip the logging
+	Skipper() bool
+	// LogRequest logs http request relateed info before making the client request
+	LogRequest(method string, url *url.URL, reqBody []byte, header http.Header)
+	// LogResponse logs http response related info after receiving the response from the server
+	LogResponse(url *url.URL, status string, resBody []byte, header http.Header, latency time.Duration)
+}
```

Comment:
why is this interface needed at all?

With the changes mentioned at the beginning, you can remove the skipper method. Then `LogRequest` and `LogResponse` can be just functions on the same `sbi_log_middlewear`